File: engine/internal/models/task.go
package models

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type TaskStatus string

const (
	TaskStatusPending   TaskStatus = "pending"
	TaskStatusRunning   TaskStatus = "running"
	TaskStatusPaused    TaskStatus = "paused"
	TaskStatusCompleted TaskStatus = "completed"
	TaskStatusFailed    TaskStatus = "failed"
)

type TaskType string

const (
	TaskTypeRecon      TaskType = "recon"
	TaskTypeScan       TaskType = "scan"
	TaskTypeFuzz       TaskType = "fuzz"
	TaskTypeExploit    TaskType = "exploit"
	TaskTypeFullChain  TaskType = "full_chain"
)

type Task struct {
	ID          uuid.UUID      `gorm:"type:uuid;primary_key" json:"id"`
	Name        string         `gorm:"size:255;not null" json:"name"`
	Type        TaskType       `gorm:"size:50;not null" json:"type"`
	Status      TaskStatus     `gorm:"size:50;not null;default:pending" json:"status"`
	TargetID    uuid.UUID      `gorm:"type:uuid;index" json:"target_id"`
	Config      JSONB          `gorm:"type:jsonb" json:"config"`
	Progress    int            `gorm:"default:0" json:"progress"`
	ResultCount int            `gorm:"default:0" json:"result_count"`
	StartedAt   *time.Time     `json:"started_at"`
	CompletedAt *time.Time     `json:"completed_at"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `gorm:"index" json:"-"`

	Target   Target       `gorm:"foreignKey:TargetID" json:"target,omitempty"`
	Results  []ScanResult `gorm:"foreignKey:TaskID" json:"results,omitempty"`
}

func (t *Task) BeforeCreate(tx *gorm.DB) error {
	if t.ID == uuid.Nil {
		t.ID = uuid.New()
	}
	return nil
}

type JSONB map[string]interface{}

File: platform/src/main/java/com/lieying/platform/model/Task.java
package com.lieying.platform.model;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Entity
@Table(name = "tasks")
public class Task {
    @Id
    @GeneratedValue
    private UUID id;

    @Column(nullable = false)
    private String name;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private TaskType type;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private TaskStatus status = TaskStatus.PENDING;

    @Column(name = "target_id")
    private UUID targetId;

    @Column(columnDefinition = "jsonb")
    private String config;

    private Integer progress = 0;

    @Column(name = "result_count")
    private Integer resultCount = 0;

    @Column(name = "started_at")
    private LocalDateTime startedAt;

    @Column(name = "completed_at")
    private LocalDateTime completedAt;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    public enum TaskType {
        RECON, SCAN, FUZZ, EXPLOIT, FULL_CHAIN
    }

    public enum TaskStatus {
        PENDING, RUNNING, PAUSED, COMPLETED, FAILED
    }
}
